### ISSUE TITLE
requestPolicyExchange in combination with offlineExchange does not send network request for first query

### ISSUE THREAD

**Issue by Mookiies** [FormidableLabs/urql#1644]
urql version & exchanges:
```
"@urql/devtools": "^2.0.3",
"@urql/exchange-graphcache": "^4.1.0",
"@urql/exchange-request-policy": "^0.1.3",
"urql": "^2.0.3",
```

Using the `requestPolicyExchange` in combination with an `offlineExchange` a network request is not sent the first time a query is executed if the cache hits.

After https://github.com/FormidableLabs/urql/pull/1641 the first operation is successfully upgraded from `cache-first` to `cache-and-network`, however no network request is actually made. I believe this is because the resulting `network-only` request is torn down before it can actually trigger a network request.

**Steps to reproduce**
(make sure to open network tab to inspect outgoing queries, look for ones to sandbox url)
1: Go to https://mookiies.github.io/request-policy-demo/ ([see code here](https://github.com/Mookiies/request-policy-demo/blob/master/src/App.js))
2: Load todos once so they are cached in offline storage
3: Refresh page
4: Load todos

**Expected behavior**
Expect a network request to be made

**Actual behavior**
Network request is never made.

Here is a video showing the issue:
https://user-images.githubusercontent.com/22333355/118325403-43d5cf80-b4b8-11eb-9266-81b1273d9af2.mov
On first toggle there is only one log output from my `shouldUpgrade` method -> no network request, but expected one
On second toggle there are two outputs -> network request

This can be a serious bug because with a more reasonable ttl of 5 minutes (my demo is 5 seconds) since data in cache may never get updated. For example in my real use case data is loaded only once at app startup, but since this first operation never results in a network request, data is never refreshed.

**Comment by Mookiies** [comment]
I copied over the urql files into my project for debugging and added logging to try and understand why this happens. I added logs in the requestPolicyExchange and in the fetch exchange. See in the `fetchExchange(teardown)` log that operation with `cache-first` is matching with operation `network-only`.
<img width="1203" alt="Screen Shot 2021-05-13 at 5 41 37 PM" src="https://user-images.githubusercontent.com/22333355/118204176-7c22d280-b412-11eb-891e-9d3e1182b310.png">
This is why I believe that the network-only request is being torn down before it can execute.


I added that log for `fetchExchange(teardown)` here in `fetchExchange`'s `filter` method
 https://github.com/FormidableLabs/urql/blob/d1a2adba7bbd7b513213b1a3118d17118456bd81/packages/core/src/exchanges/fetch.ts#L23-L26

**Comment by Mookiies** [comment]
My examples uses the issue template sandbox, but I forked it to disable cors. If it's down hopefully going to it's sandbox https://codesandbox.io/s/xenodochial-elbakyan-h82zn?file=/index.js brings it back up

**Comment by Mookiies** [comment]
I made an updated demo using the `requestPolicyExchange` from  https://github.com/FormidableLabs/urql/pull/1742 to show how that PR solves this issue and demonstrate what I think behavior should be. 

Demo: https://mookiies.github.io/request-policy-demo-fixed/ 

Source: https://github.com/Mookiies/request-policy-demo-fixed

**Comment by kitten** [comment]
fix is shipped in `@urql/exchange-request-policy@0.1.4`
